sqlx-core/src/pool/options.rs

```diff
@@ -198,6 +202,21 @@ impl<DB: Database> PoolOptions<DB> {
         self.min_connections
     }
 
+    /// Set a threshold for reporting excessive time taken to acquire a connection from
+    /// the connection pool. When the threshold is exceeded, a warning is logged.
+    ///
+    /// Defaults to a value that should not typically be exceeded by the pool enlarging
+    /// itself with an additional new connection.
+    pub fn acquire_slow_after(mut self, value: Duration) -> Self {
```

Comment:
Not everyone is going to want to log slow acquires, or want them logged at `WARN`.

IMO this should be more like [`ConnectOptions::log_slow_statements()`](https://docs.rs/sqlx/latest/sqlx/trait.ConnectOptions.html#tymethod.log_slow_statements) which takes a `LevelFilter` and a `Duration`; I'd call it `log_slow_acquires()`.

Or, to have a separate setter for the threshold (so a getter makes sense):

```rust
impl<DB: Database> PoolOptions<DB> {

    /// Log when a [`Pool::acquire()`] call takes longer than a certain threshold, indicating high load.
    ///
    /// Change the threshold duration with [`.slow_acquire_threshold()`][Self::slow_acquire_threshold]
    /// or get the current value with [`.get_slow_acquire_threshold()`][Self::get_slow_acquire_threshold].
    ///
    /// Set to [`LevelFilter::Warn`] by default. 
    /// Set to [`LevelFilter::Off`] to disable slow-acquire logging.
    pub fn log_slow_acquires(mut self, level: LevelFilter) -> Self {
        // ...
    }

    /// Set the threshold duration at which a [`Pool::acquire()`] call is considered "slow" and should be logged.
    pub fn slow_acquire_threshold(mut self, threshold: Duration) -> Self {
        // .. 
    }

    /// Get the threshold duration at which a [`Pool::acquire()`] call is considered "slow" and should be logged.
    pub fn get_slow_acquire_threshold(&self) -> Duration {
        // ..
    }
}
```

Reply:
I've added a commit to make the log level configurable.

After using first version of this PR in production a bit, I want the option to log all acquire times, so the new commit also adds that.

New default behavior:

* Only slow acquires are logged
* Slow acquires are logged at warning level

New `PoolOptions`:

* Can enable logging of all acquires by setting the log level to something other than `Off` via `log_acquires`
* Can change the log level of slow acquires to something other than `Warn` - e.g. `Error` or `Off`